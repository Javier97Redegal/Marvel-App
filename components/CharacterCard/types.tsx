type CharacterCardProps = {
    character: {
        id: number
        name: string | undefined
        thumbnail: {
            path: string
            extension: string
        }
    }
}