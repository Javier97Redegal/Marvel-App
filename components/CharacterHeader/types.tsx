export interface CharacterHeaderProps {
    id: number
}