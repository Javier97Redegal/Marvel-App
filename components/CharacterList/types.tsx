export interface CharacterListProps {
    mode: string
    query: string
}