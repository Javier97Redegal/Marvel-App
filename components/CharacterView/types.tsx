export interface CharacterViewProps {
    id: string
}