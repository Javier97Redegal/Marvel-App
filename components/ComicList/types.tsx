export interface ComicListProps {
    id: number
}