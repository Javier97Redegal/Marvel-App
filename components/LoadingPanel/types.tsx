export interface LoadingPanelProps {
    type: 'error' | 'loading' | 'idle' | 'success'
}