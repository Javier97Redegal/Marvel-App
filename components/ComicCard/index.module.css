.card {
    width: 100%;
}

.image {
    width: 100%;
    aspect-ratio: 2 / 3;
    background-size: cover;
    background-position: center;
}

.name {
    margin: 1rem 0;
    font-weight: bold;
    font-size: 1.225rem;

    @media (max-width: 1280px) {
        font-size: 1rem;        
    }
}

.date {
    font-size: .925rem;

    @media (max-width: 1280px) {
        font-size: .535rem;        
    }
}