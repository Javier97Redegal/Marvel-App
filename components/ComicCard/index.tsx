import { FC } from 'react'
import { GetComicsFromCharacterResultType } from '@/utils/MarvelAPI/AutoGeneratedTypes/GetComicsFromCharacter.types'
import styles from './index.module.css'

const ComicCard: FC<GetComicsFromCharacterResultType> = (props) => {
    const {
        title,
        thumbnail: {
            path,
            extension,
        },
        dates
    } = props

    return <>
        <div className={styles.card}>
            <div className={styles.image} style={{ backgroundImage: `url(${path}.${extension})` }}></div>
            <div className={styles.name}>{title}</div>
            <div className={styles.date}>{dates[0].date.split('-')[0]}</div>
        </div>
    </>
}

export default ComicCard