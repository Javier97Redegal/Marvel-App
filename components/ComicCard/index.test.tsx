import '@testing-library/jest-dom'
import { render, screen } from '@testing-library/react'
import ComicCard from '.'
import { GetComicsFromCharacterResultType } from '@/utils/MarvelAPI/AutoGeneratedTypes/GetComicsFromCharacter.types'

describe('ComicCard tests', () => {
    const mockComic: GetComicsFromCharacterResultType = {
        id: 1,
        title: 'Mock Comic',
        dates: [{ type: 'onsaleDate', date: '2023-01-01T00:00:00-0500' }],
        thumbnail: {
            path: 'http://example.com/image',
            extension: 'jpg'
        }
    }

    it('displays the correct title', () => {
        const { asFragment } = render(<ComicCard {...mockComic} />)

        expect(screen.getByText('Mock Comic')).toBeInTheDocument()
        expect(asFragment()).toMatchSnapshot()
    })

    it('displays the correct image', () => {
        const { container, asFragment } = render(<ComicCard {...mockComic} />)
        const imageDiv = container.querySelector('.image')

        expect(imageDiv).toHaveStyle(`background-image: url(http://example.com/image.jpg)`)
        expect(asFragment()).toMatchSnapshot()
    })

    it('displays the correct date', () => {
        const { asFragment } = render(<ComicCard {...mockComic} />)

        expect(screen.getByText('2023')).toBeInTheDocument()
        expect(asFragment()).toMatchSnapshot()
    })
})