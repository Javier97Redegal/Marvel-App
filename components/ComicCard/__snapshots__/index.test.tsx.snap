// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComicCard tests displays the correct date 1`] = `
<DocumentFragment>
  <div
    class="card"
  >
    <div
      class="image"
      style="background-image: url(http://example.com/image.jpg);"
    />
    <div
      class="name"
    >
      Mock Comic
    </div>
    <div
      class="date"
    >
      2023
    </div>
  </div>
</DocumentFragment>
`;

exports[`ComicCard tests displays the correct image 1`] = `
<DocumentFragment>
  <div
    class="card"
  >
    <div
      class="image"
      style="background-image: url(http://example.com/image.jpg);"
    />
    <div
      class="name"
    >
      Mock Comic
    </div>
    <div
      class="date"
    >
      2023
    </div>
  </div>
</DocumentFragment>
`;

exports[`ComicCard tests displays the correct title 1`] = `
<DocumentFragment>
  <div
    class="card"
  >
    <div
      class="image"
      style="background-image: url(http://example.com/image.jpg);"
    />
    <div
      class="name"
    >
      Mock Comic
    </div>
    <div
      class="date"
    >
      2023
    </div>
  </div>
</DocumentFragment>
`;
